scheduler/reconciler/reconcile_node.go

```diff
@@ -436,9 +440,9 @@ func (nr *NodeReconciler) computeForNode(
 				dstate.AutoPromote = tg.Update.AutoPromote
 				dstate.ProgressDeadline = tg.Update.ProgressDeadline
 			}
+			dstate.DesiredTotal = len(eligibleNodes)
 		}
 
-		dstate.DesiredTotal = len(eligibleNodes)
```

Comment:
We're moving this up into the `!existingDeployment` because we don't want to overwrite it on a subsequent eval. But can't the number of eligible nodes change between evals?

Reply:
It can, but we should not be setting `DesiredTotal` if there's an existing deployment, regardless.